packages/chat-client/src/utils/jwtAuth.ts

```diff
@@ -0,0 +1,107 @@
+import { base58btc } from "multiformats/bases/base58";
+import bs58 from "bs58";
+import * as ed25519 from "@noble/ed25519";
+
+interface JwtHeader {
+  typ: string;
+  alg: string;
+}
+
+export const DAY_IN_MS = 86400 * 1000;
+
+export const DID_METHOD_KEY = "key";
+export const DID_DELIMITER = ":";
+export const DID_PREFIX = "did";
+export const DID_METHOD_PKH = "pkh";
+
+export const JWT_DELIMITER = ".";
+
+export const MULTICODEC_ED25519_HEADER = "K36";
+
+export interface InviteKeyClaims {
+  iss: string;
+  sub: string;
+  aud: string;
+  pkh: string;
+  iat: number;
+  exp: number;
+}
+
+const concatUInt8Arrays = (array1: Uint8Array, array2: Uint8Array) => {
+  const mergedArray = new Uint8Array(array1.length + array2.length);
+  mergedArray.set(array1);
+  mergedArray.set(array2, array1.length);
+
+  return mergedArray;
+};
+
+export const encodeDidPkh = (accountId: string) => {
+  return `${DID_PREFIX}${DID_DELIMITER}${DID_METHOD_PKH}${DID_DELIMITER}${accountId}`;
+};
+
+export const jwtExp = (issuedAt: number) => {
+  return issuedAt + DAY_IN_MS;
+};
+
+const objectToHex = (obj: unknown) => {
+  return Buffer.from(new TextEncoder().encode(JSON.stringify(obj))).toString(
```

Comment:
This will throw in an unhandled way if `obj` is not a valid object, which `unknown` currently permits.

Reply:
@devceline. seems like a good use for `isValidObject` from utils